app/assets/javascripts/repository_columns/index.js

```diff
@@ -0,0 +1,247 @@
+  //         $('.dnd-error').remove();
+  //         $('#manageRepositoryColumn ').find('.bootstrap-tagsinput').after(
+  //           "<i class='dnd-error'>" + message + "</i>"
+  //         );
+  //       } else {
+  //         var field = { "name": xhr.responseJSON.message };
+  //         $(form).renderFormErrors('repository_column', field, true, e);
+  //       }
+  //     } else {
+  //       HelperModule.flashAlertMsg(xhr.responseJSON.message, 'danger');
+  //     }
+  //   });
+  //   if (modalID) {
+  //     form.submit();
+  //   }
+  // }
+
+  // @TODO refactor that
+  function initEditCoumnModal() {
+    var modalID = '#manageRepositoryColumn';
+    // var colRadID = '#repository_column_data_type_repositorylistvalue';
+    // var tagsInputID = '[data-role="tagsinput"]';
+    // var formID = '[data-role="manage-repository-column-form"]';
+
+    $('[data-action="edit"]').off('click').on('click', function() {
+      var editUrl = $(this).closest('li').attr('data-edit-url');
+      $.get(editUrl, function(data) {
+        $(data.html).appendTo('body').promise().done(function() {
+          $(modalID).modal('show').promise().done(function() {
+            $(modalID).on('hidden.bs.modal', function() {
+              // remove edit modal window
+              $(modalID).remove();
+              $('.modal-backdrop').remove();
+            });
+
+            initColumnTypeSelector();
+            $('#repository_column_data_type').val($(modalID).attr('data-edit-type')).trigger('click');
+            $('#repository_column_data_type').prop('disabled', true);
+            setTimeout(function() {
+              $('#repository_column_name').focus();
+            }, 500);
+
+            // if ($(modalID).attr('data-edit-type') === 'RepositoryListValue') {
+            //   var values = JSON.parse($(tagsInputID).attr('data-value'));
+            //   $('#repository_column_data_type').val('RepositoryListValue');
+            //   $(colRadID).click().promise().done(function() {
+            //     $.each(tagsInputValues, function(index, element) {
+            //       $(tagsInputID).tagsinput('add', element);
+            //     });
+            //   });
+            // }
+
+            $('[data-action="save"]').on('click', function() {
+              // if ($('#repository_column_data_type').val() === 'RepositoryListValue') {
+              //   $('#list_items').val($(tagsInputID).val());
+              // }
+              //
+              // processResponse($(formID), 'update', modalID);
+            });
+          });
+        });
+      }).fail(function() {
+        HelperModule.flashAlertMsg(
+          I18n.t('libraries.repository_columns.no_permissions'), 'danger'
+        );
+      });
+    });
+  }
+
+  function removeElementFromDom(column) {
+    $('.list-group-item[data-id="' + column.id + '"]').remove();
+    if ($('.list-group-item').length === 0) {
+      location.reload();
```

Comment:
Why we reload page if we don't have elements?

Reply:
To show ''no-columns" element.